### ISSUE TITLE
Nullish coalescing in condition doesn't affect inferred type

### ISSUE THREAD

**Issue by noahbald** [microsoft/TypeScript#47759]
# Bug Report

### 🔎 Search Terms

Nullish coalescing, inferred type, condition, ??

### 🕗 Version & Regression Information

- This is a type error
- v4.5.4

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhjAvDA2gRgDQwK5gCYCmAZgJZgF5ZjYA2NAuigLJxQAWAdAE4i54AULdtzj4QAW34BKGACoYAZin0AUCSIx+-BAH4dManRkBCRMkM0ZAbxUw7MUJBA0CHGiADm2mAFoYaKQBuFQBfIA)

### 💻 Code

```ts
const a = [1, undefined, null][Math.round(Math.random() * 3)];
if ((a ?? null) !== null) {
  console.log(a - 1);
}
```

### 🙁 Actual behavior

Error on line 3 (2533)
Object is possibly 'null' or 'undefined'.

### 🙂 Expected behavior

The inferred type for `a` should just be `number` since the condition excludes nullish values. Therefore, no error should be given.

**Comment by MartinJohns** [comment]
Sounds like a duplicate of #45941. Why would you ever write code like this?

**Comment by fatcerberus** [comment]
I'm figuring `(a ?? null) != null` is meant to be shorthand for `a !== null && a !== undefined`.

`a != null` is more idiomatic, shorter, *does* get narrowed properly, and is probably the only socially acceptable use of `!=` in JS 😄

**Comment by RyanCavanaugh** [comment]
This isn't a narrowing pattern we have detection for, and doesn't seem like a common way to test this condition.
